black-jack-redo.rb

```diff
@@ -25,88 +25,205 @@ def print_with_stars(string)
+		user_hand = user_hand + "Your cards add up to #{hand_value} points!" + "\n"
+	else
+		user_hand = user_hand + "The dealers cards add up to #{hand_value} points!" + "\n"
 	end
+	return user_hand, hand_value
 end
 
-def draw(array)
-	number = array.length - 1
-	puts array[number]
-	return array[number]
+def get_value_of_hand(current_hand)
+	user_hand, hand_value = viewhand(current_hand, 'true')
+	return hand_value
 end
 
-def drawone(array, cards, count)
-	bust = false
-	number = rand(52)
-	if array[number][:dealt] == true
-		draw(array, cards)
+def who_wins(array, current_hand, dealer_hand)
+	string = ''
+	user_hand, hand_value = viewhand(current_hand, 'true')
+	updated_dealer_hand, dealer_value = viewhand(dealer_hand, 'false')
+	if (dealer_value < 17) && (dealer_value < hand_value)
+		updated__dealer_hand = ''
+		updated_dealer_hand, dealer_value = viewhand(dealer_hand, 'false')
+		print_with_stars(updated_dealer_hand)
+		while (dealer_value < 17) && (dealer_value < hand_value)
+			updated_dealer_hand, bool = drawone(array, dealer_hand)
+			array.pop
+			updated_dealer_hand, dealer_value = viewhand(dealer_hand, 'false')
+			print_with_stars(updated_dealer_hand)
+			if bool == true
+				string = string + 'The dealer bust! You win!'
+				print_with_stars(string)
+				return
+			else
+			end
+		end
+
+		string = user_hand + updated_dealer_hand
+		if hand_value == dealer_value
+			string = string + "You and the dealer both had #{dealer_value}, it's a tie!"
+		elsif hand_value < dealer_value
+			string = string + "The dealer had #{dealer_value} and you had #{hand_value}, you lose!"
+		else hand_value > dealer_value
+			string = string + "The dealer had #{dealer_value} and you had #{hand_value}, you win!"
+		end
 	else
-		array[number][:dealt] = true
-		puts 'Your were dealt: '
-		puts array[number][:card] + ' of ' + array[number][:suit]
-		value = array[number][:value] + count
-		puts "Your cards add up to #{value} points!"
-		if value > 21
-			puts "Oh no! Your total is now #{value} which is greater than 21, you lose!"
-			bust = true
+		string = user_hand + updated_dealer_hand
+		if hand_value == dealer_value
+			string = string + "You and the dealer both had #{dealer_value}, it's a tie!"
+		elsif hand_value < dealer_value
+			string = string + "The dealer had #{dealer_value} and you had #{hand_value}, you lose!"
+		else hand_value > dealer_value
+			string = string + "The dealer had #{dealer_value} and you had #{hand_value}, you win!"
+		end
+	end
+
+	print_with_stars(string)
+end
+
+def hand_contains_ace(current_hand)
+	if (current_hand.include?('SA')) || (current_hand.include?('HA')) || (current_hand.include?('DA') )|| (current_hand.include?('CA'))
+		return true
+	else
+		return false
+	end
+end
+
+def ace_handler(current_hand)
+	array_indexes = current_hand.length
+	array_indexes.times do |array_index|
+		if current_hand[array_index] == 'SA'
+			current_hand[array_index] = 'Sa'
+		elsif current_hand[array_index] == 'HA'
```

Comment:
Condense this all down to a single statement. You're taking the `A` and changing it to an `a` in each case. You don't need four `if` evaluations to do that.